ibc-apps/ics20-transfer/types/src/denom.rs

```diff
@@ -113,6 +113,13 @@ impl Display for TracePrefix {
 pub struct TracePath(Vec<TracePrefix>);
 
 impl TracePath {
+    /// Creates a new trace path from a vector of trace prefixes.
+    /// Reverse the order of the prefixes for easier addition/removal from the end.
+    pub fn new(mut trace: Vec<TracePrefix>) -> Self {
+        trace.reverse();
+        Self(trace)
+    }
```

Comment:
Usually, `new` methods act as a constructor or initiator with the given arguments. I wouldn't expect any logic to be living under that. Can be confusing IMO. I suggest we do the `reverse()` at the point it is needed (with a comment on top).

Reply:
be0f6935

Also, I noticed `derive_more::From` is implemented already.